pyfarm/agent/entrypoints/main.py

```diff
@@ -128,15 +128,23 @@ def __init__(self):
             help="The password required to access manipulate the agent "
                  "using REST. [default: %(default)s]")
         global_network.add_argument(
-            "--systemid", config="agent_systemid", type=system_identifier,
-            default=config["agent_systemid"],
-            help="The system identification value.  This is used to help "
-                 "identify the system itself to the master when the agent "
-                 "connects. [default: %(default)s]")
-        global_network.add_argument(
-            "--systemid-cache", config="agent_systemid_cache",
-            help="The location to cache the value for --systemid. "
+            "--agent-id", config="agent_id", type=uuid_type,
+            default=AgentUUID.load(),
+            help="The UUID used to identify this agent to the master.  By "
+                 "default the agent will attempt to load a cached value "
+                 "however a specific UUID could be provided with this flag. "
                  "[default: %(default)s]")
+        global_network.add_argument(
+            "--agent-id-cache", config="agent_id_cache",
+            default=None,
+            help="The location to cache the value for --agent-uuid.  By "
+                 "default the agent's UUID will be cached in the first "
+                 "available location on the configuration path.  Declaring "
+                 "this flag will override this behavior and attempt to load "
+                 "and save the agent's UUID from this location instead.  If "
+                 "this flag is provided along with --agent-uuid the value "
+                 "loaded from the cache will be overwritten by the value "
```

Comment:
This is a bit ambiguous. Will specifying both options overwrite the UUID cache on disk?